de.fhdo.lemma.service.open_api/src/de/fhdo/lemma/service/open_api/LemmaDataSubGenerator.xtend

```diff
@@ -0,0 +1,197 @@
+package de.fhdo.lemma.service.open_api
+
+import de.fhdo.lemma.data.Context
+import de.fhdo.lemma.data.DataFactory
+import de.fhdo.lemma.data.DataField
+import de.fhdo.lemma.data.DataStructure
+import de.fhdo.lemma.data.ListType
+import io.swagger.v3.oas.models.OpenAPI
+import io.swagger.v3.oas.models.media.ArraySchema
+import io.swagger.v3.oas.models.media.Schema
+import org.apache.commons.lang3.StringUtils
+import org.slf4j.LoggerFactory
+import de.fhdo.lemma.data.DataModel
+
+/**
+ * This class is responsible for handling the generation of a LEMMA data model from an
+ * OpenAPI file in the JSON or YAML format.
+ *
+ * @author <a href="mailto:jonas.sorgalla@fh-dortmund.de">Jonas Sorgalla</a>
+ */
+class LemmaDataSubGenerator {
+    /**
+     * Map of all created DataStructures during a generation. The key contains the full-qualified
+     * name while the value contains the actual data structure created.
+     */
+    val createdDataStructures = <String, DataStructure>newHashMap
+    /**
+     * Map of all created Lists during a generation. The key contains the full-qualified
+     * name while the value contains the actual list structure created.
+     */
+    val createdListStructures = <String, ListType>newHashMap
+    /** Factory to actually create and manipulate a LEMMA DataModel */
+    val DATA_FACTORY = DataFactory.eINSTANCE
+    /** Predefined instances of the data model, version, and context
+     * which are enriched with the data structures from the OpenAPI source
+     */
+    val myDataModel = DATA_FACTORY.createDataModel
+    val myVersion = DATA_FACTORY.createVersion
+    val myContext = DATA_FACTORY.createContext
+    /** Seperator is used to build the full qualified names during the generation */
+    val seperator = "."
```

Comment:
Sep**a**rator

And: I don't think that the separator is ever going to change. For readability, I suggest to remove this constant-like field and replace its occurrence in the code with `"."`.